test/stdcpp/src/array.d

```diff
@@ -0,0 +1,57 @@
+module test.stdcpp.array;
+
+import core.stdcpp.array;
+
+unittest
+{
+    array!(int, 5) arr;
+    arr[] = [1, 2, 3, 4, 5];
+
```

Comment:
Add `arr.front++; assert(arr.front == 2);` to verify that return by `ref` works as expected.